### PR TITLE
Refactor CodeTrans with GenAIComps Microservices

### PR DESCRIPTION
**Pull request by letonghan**
## Type of Change

feature

## Description

Refactor CodeTrans with GenAIComps Microservices

## How has this PR been tested?

Local tested on both Xeon and Gaudi2.
image
image

## Dependency Change?

None.

### CODE REVIEW

**Comment by chensuyue** [comment]
Please follow the ChatQnA add test workflow, and I will help on the GHA setup.    
https://github.com/opea-project/GenAIExamples/tree/main/ChatQnA/tests

File: CodeTrans/microservice/gaudi/README.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+```bash
+git clone https://github.com/opea-project/GenAIComps.git
+cd GenAIComps
+```
+
+### 2. Build the LLM Docker Image with the following command
+
+```bash
+docker build -t opea/gen-ai-comps:llm-tgi-gaudi-server --no-cache --build-arg https_proxy=$https_proxy --build-arg http_proxy=$http_proxy -f comps/llms/langchain/docker/Dockerfile .
```
Comment (chensuyue) [inline CodeTrans/microservice/gaudi/README.md:19]: `opea/gen-ai-comps:llm-tgi-server` use the same name as below.

File: CodeTrans/microservice/gaudi/README.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+```
+
+Then run the command `docker images`, you will have the following Docker Image:
+
+- `opea/gen-ai-comps:llm-tgi-server`
+
+### 3. Build MegaService Docker Image
+
+```bash
+docker build -t opea/gen-ai-comps:codetrans-megaservice-server --build-arg https_proxy=$https_proxy --build-arg http_proxy=$http_proxy -f docker/Dockerfile .
```
Comment (chensuyue) [inline CodeTrans/microservice/gaudi/README.md:29]: git clone https://github.com/opea-project/GenAIExamples
cd GenAIExamples/CodeTrans/microservice/gaudi/  

where is the dockerfile?

File: CodeTrans/microservice/gaudi/README.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+```
+
+### 4. Build UI Docker Image
+
+```bash
+cd ../../ui
```
Comment (chensuyue) [inline CodeTrans/microservice/gaudi/README.md:35]: where is the dockerfile？

**Review by lvliang-intel — APPROVED** [review]

File: ChatQnA/tests/test_chatqna_on_gaudi.sh [hunk 4]
```diff
@@ -6,8 +6,7 @@ set -x
 
 WORKPATH=$(dirname "$PWD")
 LOG_PATH="$WORKPATH/tests"
-ip_name=$(echo $(hostname) | tr '[a-z]-' '[A-Z]_')_$(echo 'IP')
-ip_address=$(eval echo '$'$ip_name)
+ip_address=$(hostname -I | awk '{print $1}')
```
Comment (chensuyue) [inline ChatQnA/tests/test_chatqna_on_gaudi.sh:9]: ip_address is a secrets in GHA, so it is correct.

Comment (letonghan) [reply ChatQnA/tests/test_chatqna_on_gaudi.sh:9]: got it

**Review by chensuyue — APPROVED** [review]

**Pull request merged**